File: samples/database-migrations/DatabaseMigrations.AppHost/AppHost.cs
var builder = DistributedApplication.CreateBuilder(args);

var sqlserver = builder.AddSqlServer("sqlserver")
    .WithDataVolume()
    .WithLifetime(ContainerLifetime.Persistent);

var db1 = sqlserver.AddDatabase("db1");

var migrationService = builder.AddProject<Projects.DatabaseMigrations_MigrationService>("migration")
    .WithReference(db1)
    .WaitFor(db1);

builder.AddProject<Projects.DatabaseMigrations_ApiService>("api")
    .WithReference(db1)
    .WaitForCompletion(migrationService);

builder.Build().Run();

File: samples/database-migrations/DatabaseMigrations.AppHost/apphost.ts
import { createBuilder, ContainerLifetime } from "./.modules/aspire.js";

const builder = await createBuilder();

const sqlserver = await builder.addSqlServer("sqlserver")
    .withDataVolume()
    .withLifetime(ContainerLifetime.Persistent);

const db1 = sqlserver.addDatabase("db1");

// POLYGLOT GAP: AddProject<Projects.DatabaseMigrations_MigrationService>("migration") — generic type parameter for project reference is not available.
const migrationService = builder.addProject("migration")
    .withReference(db1)
    .waitFor(db1);

// POLYGLOT GAP: AddProject<Projects.DatabaseMigrations_ApiService>("api") — generic type parameter for project reference is not available.
// POLYGLOT GAP: .WaitForCompletion(migrationService) — WaitForCompletion is not available in the TypeScript polyglot SDK; only WaitFor is available.
const api = builder.addProject("api")
    .withReference(db1)
    .waitFor(migrationService);

await builder.build().run();
